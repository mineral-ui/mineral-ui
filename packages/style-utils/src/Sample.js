/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */
import React from 'react';
import { createStyledComponent } from './index';

const Root = createStyledComponent('span', (props, theme) => ({
  backgroundColor: theme.Sample_backgroundColor || 'lavender',
  color: theme.Sample_color || theme.color_primary,
  display: 'inline-block',
  outline: `1px solid ${theme.color_border}`,
  padding: theme.spacing_single
}));

type Props = {|
  className?: string,
  text?: string
|};

export default function Sample({ className, text = 'Sample' }: Props) {
  return <Root className={className}>{text}</Root>;
}
