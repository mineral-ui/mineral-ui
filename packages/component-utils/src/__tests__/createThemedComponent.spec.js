/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */
import React from 'react';
import { mount } from 'enzyme';
import { createThemedComponent, ThemeProvider } from '../index';
import Sample from '../__demo__/components/Sample';

function mountThemedSample() {
  const ThemedSample = createThemedComponent(Sample, {
    color_primary: 'mediumvioletred'
  });

  return mount(
    <ThemeProvider>
      <ThemedSample />
    </ThemeProvider>
  );
}

describe('createThemedComponent', () => {
  it('renders correctly', () => {
    const themedSample = mountThemedSample();

    expect(themedSample).toMatchSnapshot();
  });
});
