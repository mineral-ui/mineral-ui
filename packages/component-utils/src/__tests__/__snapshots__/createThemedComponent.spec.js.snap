// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createThemedComponent renders correctly 1`] = `
.css-m5004x,
[data-css-m5004x] {
  background-color: lavender;
  color: mediumvioletred;
  display: inline-block;
  outline: 1px solid #afbacc;
  padding: 1em;
}

<ThemeProvider>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor_danger": "#db2929",
        "backgroundColor_danger_active": "#f27474",
        "backgroundColor_danger_focus": "#db2929",
        "backgroundColor_danger_hover": "#eb4d4d",
        "backgroundColor_disabled": "#dde3ed",
        "backgroundColor_input_danger": "#fce3e3",
        "backgroundColor_input_success": "#e8fcf2",
        "backgroundColor_input_warning": "#fcf1dc",
        "backgroundColor_link_focus": "#dde3ed",
        "backgroundColor_success": "#10a35a",
        "backgroundColor_success_active": "#40db8d",
        "backgroundColor_success_focus": "#10a35a",
        "backgroundColor_success_hover": "#1fc06f",
        "backgroundColor_warning": "#e3a322",
        "backgroundColor_warning_active": "#f0b241",
        "backgroundColor_warning_focus": "#e3a322",
        "backgroundColor_warning_hover": "#f0b241",
        "borderColor": "#afbacc",
        "borderColor_active": "#4a89e8",
        "borderColor_danger": "#ad0e0e",
        "borderColor_danger_active": "#7d0606",
        "borderColor_danger_focus": "#7d0606",
        "borderColor_danger_hover": "#db2929",
        "borderColor_focus": "#164ea8",
        "borderColor_hover": "#2e6fd9",
        "borderColor_success": "#0a8f4c",
        "borderColor_success_active": "#046132",
        "borderColor_success_focus": "#046132",
        "borderColor_success_hover": "#1fc06f",
        "borderColor_warning": "#d19411",
        "borderColor_warning_active": "#916504",
        "borderColor_warning_focus": "#916504",
        "borderColor_warning_hover": "#e3a322",
        "borderRadius_1": "0.5625em",
        "borderRadius_2": "0.75em",
        "borderRadius_3": "1.125em",
        "borderRadius_4": "1.5em",
        "borderRadius_5": "2.25em",
        "color_black": "#000",
        "color_caption": "#58606e",
        "color_gray_10": "#f5f7fa",
        "color_gray_100": "#333840",
        "color_gray_20": "#ebeff5",
        "color_gray_30": "#dde3ed",
        "color_gray_40": "#c8d1e0",
        "color_gray_50": "#afbacc",
        "color_gray_60": "#8e99ab",
        "color_gray_70": "#707a8a",
        "color_gray_80": "#58606e",
        "color_gray_90": "#434a54",
        "color_heading": "#164ea8",
        "color_helpText_danger": "#ad0e0e",
        "color_helpText_success": "#0a8f4c",
        "color_helpText_warning": "#a67305",
        "color_link": "#114091",
        "color_link_active": "#0f397d",
        "color_link_focus": "#0f397d",
        "color_link_hover": "#1f5dc2",
        "color_placeholder": "#8e99ab",
        "color_text": "#333840",
        "color_text_danger": "#940909",
        "color_text_disabled": "#afbacc",
        "color_text_ondanger": "white",
        "color_text_onprimary": "white",
        "color_text_onsuccess": "white",
        "color_text_onwarning": "white",
        "color_text_primary": "#114091",
        "color_text_success": "#06783f",
        "color_text_warning": "#a67305",
        "color_theme_10": "#e6eefc",
        "color_theme_100": "#0f397d",
        "color_theme_20": "#c2dbfc",
        "color_theme_30": "#9dc2fa",
        "color_theme_40": "#72a5f2",
        "color_theme_50": "#4a89e8",
        "color_theme_60": "#2e6fd9",
        "color_theme_70": "#1f5dc2",
        "color_theme_80": "#164ea8",
        "color_theme_90": "#114091",
        "color_white": "#fff",
        "direction": "ltr",
        "fontFamily": "Open Sans",
        "fontFamily_monospace": "monospace",
        "fontFamily_system": "-apple-system, system-ui, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
        "fontSize_base": "8px",
        "fontSize_h1": "4.25em",
        "fontSize_h2": "3.5em",
        "fontSize_h3": "2.75em",
        "fontSize_h4": "2.25em",
        "fontSize_h5": "1.75em",
        "fontSize_h6": "1.75em",
        "fontSize_mouse": "1.375em",
        "fontSize_prose": "2em",
        "fontSize_ui": "1.75em",
        "fontWeight_bold": 700,
        "fontWeight_extraBold": 800,
        "fontWeight_regular": 400,
        "fontWeight_semiBold": 600,
        "lineHeight": 1.25,
        "lineHeight_prose": 1.5,
        "shadow_1": "rgba(0,0,0,0.5) 0 1px 2px",
        "shadow_2": "rgba(0,0,0,0.5) 0 1px 1px, rgba(0,0,0,0.35) 0 2px 4px",
        "shadow_3": "rgba(0,0,0,0.5) 0 1px 1px, rgba(0,0,0,0.25) 0 2px 8px",
        "shadow_4": "rgba(0,0,0,0.5) 0 1px 1px, rgba(0,0,0,0.30) 0 4px 10px",
        "shadow_5": "rgba(0,0,0,0.5) 0 1px 2px, rgba(0,0,0,0.60) 0 15px 15px",
        "size_large": "5em",
        "size_medium": "4em",
        "size_small": "3em",
        "spacing_double": "2em",
        "spacing_half": "0.5em",
        "spacing_oneAndAHalf": "1.5em",
        "spacing_quad": "4em",
        "spacing_quarter": "0.25em",
        "spacing_single": "1em",
        "spacing_triple": "3em",
        "zIndex_100": 100,
        "zIndex_1600": 1600,
        "zIndex_200": 200,
        "zIndex_400": 400,
        "zIndex_800": 800,
      }
    }
  >
    <ThemedComponent>
      <Themed(Sample)
        theme={
          Object {
            "backgroundColor_danger": "#db2929",
            "backgroundColor_danger_active": "#f27474",
            "backgroundColor_danger_focus": "#db2929",
            "backgroundColor_danger_hover": "#eb4d4d",
            "backgroundColor_disabled": "#dde3ed",
            "backgroundColor_input_danger": "#fce3e3",
            "backgroundColor_input_success": "#e8fcf2",
            "backgroundColor_input_warning": "#fcf1dc",
            "backgroundColor_link_focus": "#dde3ed",
            "backgroundColor_success": "#10a35a",
            "backgroundColor_success_active": "#40db8d",
            "backgroundColor_success_focus": "#10a35a",
            "backgroundColor_success_hover": "#1fc06f",
            "backgroundColor_warning": "#e3a322",
            "backgroundColor_warning_active": "#f0b241",
            "backgroundColor_warning_focus": "#e3a322",
            "backgroundColor_warning_hover": "#f0b241",
            "borderColor": "#afbacc",
            "borderColor_active": "#4a89e8",
            "borderColor_danger": "#ad0e0e",
            "borderColor_danger_active": "#7d0606",
            "borderColor_danger_focus": "#7d0606",
            "borderColor_danger_hover": "#db2929",
            "borderColor_focus": "#164ea8",
            "borderColor_hover": "#2e6fd9",
            "borderColor_success": "#0a8f4c",
            "borderColor_success_active": "#046132",
            "borderColor_success_focus": "#046132",
            "borderColor_success_hover": "#1fc06f",
            "borderColor_warning": "#d19411",
            "borderColor_warning_active": "#916504",
            "borderColor_warning_focus": "#916504",
            "borderColor_warning_hover": "#e3a322",
            "borderRadius_1": "0.5625em",
            "borderRadius_2": "0.75em",
            "borderRadius_3": "1.125em",
            "borderRadius_4": "1.5em",
            "borderRadius_5": "2.25em",
            "color_black": "#000",
            "color_caption": "#58606e",
            "color_gray_10": "#f5f7fa",
            "color_gray_100": "#333840",
            "color_gray_20": "#ebeff5",
            "color_gray_30": "#dde3ed",
            "color_gray_40": "#c8d1e0",
            "color_gray_50": "#afbacc",
            "color_gray_60": "#8e99ab",
            "color_gray_70": "#707a8a",
            "color_gray_80": "#58606e",
            "color_gray_90": "#434a54",
            "color_heading": "#164ea8",
            "color_helpText_danger": "#ad0e0e",
            "color_helpText_success": "#0a8f4c",
            "color_helpText_warning": "#a67305",
            "color_link": "#114091",
            "color_link_active": "#0f397d",
            "color_link_focus": "#0f397d",
            "color_link_hover": "#1f5dc2",
            "color_placeholder": "#8e99ab",
            "color_text": "#333840",
            "color_text_danger": "#940909",
            "color_text_disabled": "#afbacc",
            "color_text_ondanger": "white",
            "color_text_onprimary": "white",
            "color_text_onsuccess": "white",
            "color_text_onwarning": "white",
            "color_text_primary": "#114091",
            "color_text_success": "#06783f",
            "color_text_warning": "#a67305",
            "color_theme_10": "#e6eefc",
            "color_theme_100": "#0f397d",
            "color_theme_20": "#c2dbfc",
            "color_theme_30": "#9dc2fa",
            "color_theme_40": "#72a5f2",
            "color_theme_50": "#4a89e8",
            "color_theme_60": "#2e6fd9",
            "color_theme_70": "#1f5dc2",
            "color_theme_80": "#164ea8",
            "color_theme_90": "#114091",
            "color_white": "#fff",
            "direction": "ltr",
            "fontFamily": "Open Sans",
            "fontFamily_monospace": "monospace",
            "fontFamily_system": "-apple-system, system-ui, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
            "fontSize_base": "8px",
            "fontSize_h1": "4.25em",
            "fontSize_h2": "3.5em",
            "fontSize_h3": "2.75em",
            "fontSize_h4": "2.25em",
            "fontSize_h5": "1.75em",
            "fontSize_h6": "1.75em",
            "fontSize_mouse": "1.375em",
            "fontSize_prose": "2em",
            "fontSize_ui": "1.75em",
            "fontWeight_bold": 700,
            "fontWeight_extraBold": 800,
            "fontWeight_regular": 400,
            "fontWeight_semiBold": 600,
            "lineHeight": 1.25,
            "lineHeight_prose": 1.5,
            "shadow_1": "rgba(0,0,0,0.5) 0 1px 2px",
            "shadow_2": "rgba(0,0,0,0.5) 0 1px 1px, rgba(0,0,0,0.35) 0 2px 4px",
            "shadow_3": "rgba(0,0,0,0.5) 0 1px 1px, rgba(0,0,0,0.25) 0 2px 8px",
            "shadow_4": "rgba(0,0,0,0.5) 0 1px 1px, rgba(0,0,0,0.30) 0 4px 10px",
            "shadow_5": "rgba(0,0,0,0.5) 0 1px 2px, rgba(0,0,0,0.60) 0 15px 15px",
            "size_large": "5em",
            "size_medium": "4em",
            "size_small": "3em",
            "spacing_double": "2em",
            "spacing_half": "0.5em",
            "spacing_oneAndAHalf": "1.5em",
            "spacing_quad": "4em",
            "spacing_quarter": "0.25em",
            "spacing_single": "1em",
            "spacing_triple": "3em",
            "zIndex_100": 100,
            "zIndex_1600": 1600,
            "zIndex_200": 200,
            "zIndex_400": 400,
            "zIndex_800": 800,
          }
        }
      >
        <ThemeProvider
          theme={
            Object {
              "color_primary": "mediumvioletred",
            }
          }
        >
          <ThemeProvider
            theme={
              Object {
                "color_primary": "mediumvioletred",
              }
            }
          >
            <Sample>
              <glamorous(span)>
                <span
                  className="css-m5004x"
                >
                  Sample
                </span>
              </glamorous(span)>
            </Sample>
          </ThemeProvider>
        </ThemeProvider>
      </Themed(Sample)>
    </ThemedComponent>
  </ThemeProvider>
</ThemeProvider>
`;
